modules/bene_promo_redirect/bene_promo_redirect.module

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/**
+ * @file
+ * Bene Promo Redirect code.
+ */
+
+use Drupal\Component\Utility\UrlHelper;
+
+/**
+ * Implements hook_page_attachments().
+ */
+function bene_promo_redirect_page_attachments(array &$page) {
+  $config = \Drupal::config('bene_promo_redirect.settings');
+  // Only redirect when enabled:
+  if ($config->get('enabled')) {
+    $destination = $config->get('dest');
+    if (!empty($destination) && UrlHelper::isValid($destination)) {
+      $route = \Drupal::routeMatch()->getRouteObject();
+      // Don't redirect admin pages:
+      if (!\Drupal::service('router.admin_context')->isAdminRoute($route)) {
+        $request = \Drupal::request();
+        $current_uri = $request->getRequestUri();
+        // Don't redirect User pages:
+        if (strpos($current_uri, "/user") !== 0) {
+          // Now let's match against our path specifications:
+          if ($config->get('redirect_all')) {
+            $pathmatch = TRUE;
+          }
+          else {
+            $pathmatch = FALSE;
+            foreach ($config->get('paths') as $path) {
+              if (in_array($current_uri, [$path, "/$path"])) {
+                $pathmatch = TRUE;
+                continue;
+              }
+            }
+          }
+          if ($pathmatch) {
+            $page['#attached']['library'][] = 'bene_promo_redirect/bene-promo-redirect';
+            // Include domains to match against for use in JS:
+            $destination_parts = parse_url($config->get('dest'));
+            $page['#attached']['drupalSettings']['benePromoRedirect'] = [
```

Comment:
You probably need to include the cache tag for the config here so the pages get cleared when the config is updated. Not sure if it'll be an issue with this, but on Feeding America we had to disable the "page_cache" (internal page cache) module to get the functionality working. I'm waiting on a response from Pantheon about the impacts of that on the Global CDN. Doesn't seem like it should impact it since page_cache is for storing the data in the db or redis, but we'll see what they say.

Reply:
FWIW, Pantheon confirmed the page_cache module shouldn't be needed since the CDN is used for caching anonymous page views.

Reply:
I thought the same thing and then tested and it didn't seem necessary. Re-tested just now and I got different results, so adding this!